tips/tests/tips_views_tests.py

```diff
@@ -0,0 +1,90 @@
+import pytest
+from django.utils import timezone
+from django.shortcuts import get_object_or_404
+from tips.models import Tip
+from django.template import TemplateDoesNotExist
+from django.test.client import RequestFactory
+from tips.forms import TipForm
+from tips.views import board
+from django.template.loader import get_template
+
+AUTHOR = 'RICK'
+OTHER_AUTHOR = 'RICKEN'
+CATEGORY = 'Clothing'
+DATE = timezone.now()
+TEXT = 'I BOUGHT A TESLA IN 50$ DISCOUNT IN LEVINSKI MARKET'
+
+FORM_DATA = {'author': OTHER_AUTHOR, 'text': TEXT, 'category': CATEGORY}
+
+
+@pytest.fixture
+def generate_tip():
+    return Tip.create_tip(
+        category=CATEGORY,
+        author=AUTHOR,
+        date=DATE,
+        text=TEXT,
+    )
+
+
+@pytest.fixture
+def request_factory():
+    return RequestFactory()
+
+
+@pytest.fixture
+def generate_form(request_factory):
+    post_request = request_factory.post('tips/add_tip', FORM_DATA,)
+    form = TipForm(post_request.POST)
+    return form
+
+
+@pytest.fixture
+def generate_get_request(request_factory):
+    get_request = request_factory.get('tips/')
+    return get_request
+
+
+@pytest.mark.django_db
+class TestTipForms:
+    def test_form(self, db, generate_form, generate_tip):
+        form = generate_form
+        assert form.is_valid()
+
+    def test_add_tip(self, generate_form):
+        form = generate_form
+        tip = form.save()
+        assert tip in Tip.objects.all()
+
+    def test_edit_tip(self, generate_form, generate_tip):
+        generate_tip.save()
+        tip = get_object_or_404(Tip, pk=generate_tip.id)
+        assert tip.author == AUTHOR
+        form = generate_form
+        tip = form.save(commit=False)
+        tip.save()
+        assert tip.author == OTHER_AUTHOR
+
```

Comment:
What happens if you edit a Tip and you delete the description? Does it work? You should add a test to validate that.

Reply:
It's like in add_tip because the form is used